src/workflows/airqo_etl_utils/date.py

```diff
@@ -67,59 +67,59 @@ def get_dag_date_time_values(
 
     @staticmethod
     def get_query_date_time_values(hours=1, days=0):
-        start_date_time = datetime.utcnow() - timedelta(hours=hours)
+        start_date_time = datetime.now(timezone.utc) - timedelta(hours=hours)
         end_date_time = start_date_time + timedelta(hours=hours)
 
         if days != 0:
-            start_date_time = datetime.utcnow() - timedelta(days=days)
+            start_date_time = datetime.now(timezone.utc) - timedelta(days=days)
             end_date_time = start_date_time + timedelta(days=days)
 
         return date_to_str_hours(start_date_time), date_to_str_hours(end_date_time)
 
 
 def get_utc_offset_for_hour(subject_hour: int) -> int:
-    hour = datetime.utcnow().hour
+    hour = datetime.now(timezone.utc).hour
     if hour < subject_hour:
         return abs(hour - subject_hour)
     elif hour > subject_hour:
         return subject_hour - hour
     return hour
 
 
-def predict_str_to_date(st: str):
+def predict_str_to_date(st: str) -> datetime:
     """
     Converts a predict string to utc datetime
     """
 
     st = st.replace(" GMT", "")
     date_time = datetime.strptime(st, "%a, %d %b %Y %H:%M:%S")
-    date_time = date_time + timedelta(hours=3)
+    date_time = date_time.replace(tzinfo=timezone.utc) + timedelta(hours=3)
     return date_time
 
 
-def str_to_date(st: str, date_format="%Y-%m-%dT%H:%M:%SZ"):
+def str_to_date(st: str, date_format="%Y-%m-%dT%H:%M:%SZ") -> datetime:
     """
-    Converts a string to datetime
+    Converts a string to a timezone-aware datetime object set to UTC.
     """
 
     try:
-        return datetime.strptime(st, "%Y-%m-%dT%H:%M:%S.%fZ")
-    except:
-        return datetime.strptime(st, date_format)
+        return datetime.strptime(st, "%Y-%m-%dT%H:%M:%S.%fZ").replace(tzinfo=timezone.utc)
+    except ValueError:
+        return datetime.strptime(st, date_format).replace(tzinfo=timezone.utc)
```

Comment:
At this point we can log the exception.
Please do the final return in the else block.

`try:
       # Some Code.... 
except:
       # optional block
       # Handling of exception (if required)
else:
       # execute if no exception
finally:
      # Some code .....(always executed)`

Reply:
On it.